src/bika/lims/content/analysisservice.py

```diff
@@ -496,7 +496,8 @@ def getAvailableMethodUIDs(self):
         Returns the UIDs of the available methods. it is used as a
         vocabulary to fill the selection list of 'Methods' field.
         """
-        method_uids = self.getRawMethods()
+        # N.B. we return a copy of the list to avoid accidental writes
+        method_uids = map(lambda x: x, self.getRawMethods())
```

Comment:
For what I've seen, the slicing technique is a lot faster. And seems that is especially faster for lists with a few items (<1000)

```python
method_uids = self.getRawMethods()[:]
```

Reply:
Hmm, not sure:
```
In [3]: def test():
   ...:     lst = [1,2,3]
   ...:     lst1 = map(lambda x: x, lst)
   ...:

In [4]: timeit test
10000000 loops, best of 3: 61.7 ns per loop

In [5]: def test():
   ...:     lst = [1,2,3]
   ...:     lst1 = lst[:]
   ...:

In [6]: timeit test
The slowest run took 50.38 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 3: 61.5 ns per loop
```

This looks to me like it does not really perform well in the first run.

Where did you read about the slicing method?

Reply:
K, probably need to call the function:
```
In [10]: def test():
    ...:     lst = [1,2,3]
    ...:     lst1 = map(lambda x: x, lst)
    ...:

In [11]:

In [11]: timeit test()
The slowest run took 4.60 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 881 ns per loop

In [12]: def test():
    ...:     lst = [1,2,3]
    ...:     lst1 = lst[:]
    ...:

In [13]:

In [13]: timeit test()
The slowest run took 4.74 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 402 ns per loop
```

Reply:
Haven't done the timeit testing myself, my statement is based on what i've been reading from different places. I might be wrong, though. For instance: 
- https://stackoverflow.com/questions/47859482/what-is-the-fastest-way-to-make-a-shallow-copy-of-list-in-python3-5#answer-47859483
- https://stackoverflow.com/questions/2612802/list-changes-unexpectedly-after-assignment-how-do-i-clone-or-copy-it-to-prevent/2612990#2612990

Reply:
Changed it now to the slicing method as you suggested. Thanks for pointing out the links! It's the small things we need to take care of to make the system as fast as possible